Integer Palindrome Check

shrutivanw opened the pull request:
CS Fundamentals' assignment to check if the digits in the integer form a palindrome.

shrutivanw committed:
Integer Palindrome Check

CS Fundamentals' assignment to check if the digits in the integer form a palindrome.

Hamled commented on README.md:
```diff
@@ -1 +1,9 @@
+* For example, 12521 and 4554 are palindromes. 326725 is not a palindrome.
+* Share and explain the time and space complexities for your solution.
+* If you think of multiple approaches to solve the problem, implement the solution which minimizes space complexity. Explain the other approaches, and your decision in comments above the code.
+**Note**: Do not use Ruby provided functionality.
```
We should have an extra blank line above this so that the note goes into its own paragraph.

shrutivanw replied:
Done.

Hamled commented on lib/integer_palindrome_check.rb:
```diff
@@ -0,0 +1,5 @@
+# Returns true if the input positive integer number forms a palindrome. Returns false otherwise.
+def is_palindrome(number)
+  puts "NOT IMPLEMENTED"
+  return true
```
How would you feel about doing `raise NotImplementedError` here instead?

Best practice I think is generally to avoid `puts` statements and other things that will print to the console for code being tested, because that output kind of gets randomly interspersed with the test runner's output.

shrutivanw replied:
Done.

Hamled commented on specs/integer_palindrome_check_spec.rb:
```diff
@@ -0,0 +1,59 @@
+
+describe "integer palindrome check" do
+  describe "basic tests" do
+    describe "test 1" do
```
Same as the other repos, we should probably remove the `describe` blocks that are immediate parents of individual tests.

shrutivanw replied:
done!

Hamled commented on specs/integer_palindrome_check_spec.rb:
```diff
@@ -0,0 +1,59 @@
+
+  # if the parameter is an object, check for nil
+  describe "nil object input test" do
+    it "nil object is not an integer" do
```
I could see this test going into the edge cases section.

shrutivanw replied:
done!

shrutivanw committed:
Incorporating CR feedback

- Removed inner most nesting of describe blocks in test specs
- Moved nil test to be part of edge cases tests
- changed the lib to throw not implemented error
- Fixed the rendering for readme file

shrutivanw commented:
Thank you @Hamled Feedback incorporated.
